src/components/elements/annotationSidePanel/components/annotationWizard/steps/ParentClassification.tsx

```diff
@@ -0,0 +1,186 @@
+import { AnnotationTarget, ParentClass, DropdownItem, Dict } from "app/Types";
+
+/* Import Icons */
+import { faCircleExclamation } from '@fortawesome/free-solid-svg-icons';
+
+/* Import Compoinents */
+import { Button, Dropdown } from "components/elements/customUI/CustomUI";
+
+
+/* Props Type */
+type Props = {
+    index: number,
+    parentClass: ParentClass,
+    selected: boolean,
+    parentClasses: ParentClass[],
+    annotationTarget?: AnnotationTarget,
+    formValues?: Dict,
+    superClass: DigitalSpecimen | DigitalMedia | Dict,
+    SetFieldValue?: Function,
+    SetParentClasses: Function
+};
+
+
+/**
+ * Component that renders a parent class selector in the second step of the annotation wizard when a parent class is missing
+ * @param index The index of the class segment in the original targeted JSON path
+ * @param parentClass The parent class provided and to check
+ * @param selected Boolean indicating if the new instance option has been selected
+ * @param parentClasses An array of all parent classes of the targeted class / term
+ * @param annotationTarget The annotation target set
+ * @param formValues The current values in the annotation wizard form
+ * @param superClass The provided super class
+ * @param SetFieldValue Function to set a value of a field in the form
+ * @param SetParentClasses Function to set and update the parent classes state variable
+ * @returns JSX Component
+ */
+const ParentClassification = (props: Props) => {
+    const { index, parentClass, selected, parentClasses, annotationTarget, formValues, superClass, SetFieldValue, SetParentClasses } = props;
+
+    /* Hooks */
+    const dispatch = useAppDispatch();
+
+    /* Base variables */
+    const titleText: string = !selected ? `Add instance of ${parentClass.name}` : 'Currently selected';
+
+    /* Depending on the index and previous classes, check if there was no other missing parent class and if so render the appropiate select options */
+    if (index > 0 && !parentClasses[index - 1].present) {
+        return <> </>;
+    } else if (!parentClass.present) {
+        return (
+            <div className="mt-2">
+                <div className="bgc-accent-soft br-corner px-3 py-1">
+                    <p className="fs-4 fw-lightBold">
+                        <FontAwesomeIcon icon={faCircleExclamation}
+                            className="tc-grey"
+                        />
+                        {` Parent class '${parentClass.name}' is not present.
+                                Add a new instance of '${MakeJsonPathReadableString(annotationTarget?.jsonPath ?? '')}' by
+                                also adding the parent class.`
+                        }
+                    </p>
+                </div>
+
+                <Button type="button"
+                    variant="primary"
+                    disabled={selected}
+                    className="fs-5 mt-3 py-1 px-3"
+                    OnClick={() => {
```

Comment:
`onClick()` ?

Reply:
Normally yes, but this is a custom UI component of mine, and when I pass along functions to custom components I capitalize them to easily separate them from normal variables. So here it is passed as OnClick, but in the Button component it is assigned as onClick yes